modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/SpeechToTextServiceImpl.java

```diff
@@ -214,11 +220,24 @@ protected OrganizationDirectoryService getOrganizationDirectoryService() {
     return organizationDirectoryService;
   }
 
-  @Reference
+  //TODO: Improve OSGI wiring to change STT Engine implementation during OC execution
+  @Reference(
+          name = "SpeechToTextEngine",
+          policy = ReferencePolicy.STATIC,
+          policyOption = ReferencePolicyOption.GREEDY,
+          cardinality = ReferenceCardinality.MANDATORY,
+          target = "(engineType=vosk)",
+          unbind = "unsetSpeechToTextEngine",
+          bind = "setSpeechToTextEngine"
+  )
```

Comment:
```suggestion
  @Reference(
          name = "SpeechToTextEngine",
          target = "(engineType=vosk)"
  )
```

The default values for policy and cardinality are `STATIC` and `MANDATORY`. Instead of the policy option `GREEDY`, I would use the default `RELUCTANT`; if a new service becomes available, it will bind to it. This should not happen because we should have only one service that matches the target filter.